rcore/backlight.c

```diff
@@ -57,13 +57,15 @@ void rcore_backlight_init(void)
 // use the backlight as additional alert by flashing it
 void rcore_backlight_on(uint16_t brightness_pct, uint16_t time)
 {
-    backlight_message_t msg;
+    // create the message, malloc some memory for it. 
+    // We then add the pointer to this message into the queue
+    backlight_message_t *msg = malloc(sizeof(backlight_message_t));
```

Comment:
this is more correct than stacking it but also we should endeavor to not dynamically allocate at all here if we can avoid it

Reply:
ideally at some point i'll change it to use a dma lookup table for the fadeout. Most of this mess can go then as we can switch to a single semaphore and a single message. No need to post the on time and whatnot through a queue. Simplify